### PR TITLE
upload url refactoring

### PR DESCRIPTION
**Pull request by magaofei**
### What was a problem?

```
upload url refactoring
All test completed
```

### How this PR fixes the problem?

```
DESCRIBE HERE
```

### Check lists (check `x` in `[ ]` of list items)

- [x] Run MobSF unit tests
- [ ] Tested Working on Linux, Mac, and Windows
- [x] Coding style (indentation, etc)

### Additional Comments (if any)

```
DESCRIBE HERE
```

### CODE REVIEW

File: MobSF/rest_api.py [hunk 1]
```diff
@@ -41,15 +42,12 @@ def api_auth(meta):
...
-            response = make_api_response(resp, 500)
-        else:
-            response = make_api_response(resp)
+    if request.method == POST:
+        upload = Upload(request)
+        return upload.upload_api()
     else:
-        response = make_api_response({"error": "Method Not Allowed"}, 405)
-    return response
+        return HttpResponseNotAllowed([POST])
```
[the remark this answers is not in the document]
Comment (magaofei) [reply MobSF/rest_api.py]: It return empty. Only 405 status code

File: MobSF/utils.py [hunk 2]
```diff
@@ -606,3 +606,28 @@ def check_basic_env():
...
+    def is_allow_file(self):
+        """
+        return bool
+        """
+        if self.is_apk() or self.is_zip() or self.is_ipa() or self.is_appx():
+            return True
+        return False
+    
+    def is_apk(self):
+        return ((self.file_type in settings.APK_MIME) and self.file_name_lower.endswith('.apk'))
```
[the remark this answers is not in the document]
Comment (magaofei) [reply MobSF/utils.py]: ok

File: MobSF/views.py [hunk 3]
```diff
@@ -29,22 +30,11 @@
...
-    """
-    try:
-        db_obj = RecentScansDB.objects.filter(MD5=md5)
-        if not db_obj.exists():
-            new_db_obj = RecentScansDB(
-                NAME=name, MD5=md5, URL=url, TS=timezone.now())
-            new_db_obj.save()
-    except:
-        PrintException("[ERROR] Adding Scan URL to Database")
+from .forms import UploadFileForm, FormUtil
```
[the remark this answers is not in the document]
Comment (magaofei) [reply MobSF/views.py]: ok

File: MobSF/scanning.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+        data = {
+            'url': 'StaticAnalyzer/?name={}&type=apk&checksum={}'.format(self.file_name, md5),
+            'status': 'success'
+        }
+
+    
+        add_to_recent_scan(self.file_name, md5, data['url'])
+        
+        data["hash"] = md5
```
[the remark this answers is not in the document]
Comment (magaofei) [reply MobSF/scanning.py]: to empty ? 
the api response need `scan_type` `file_name` `hash`, and the template response need `url` `status`. I just put them in a dict.

File: MobSF/rest_api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajinabraham) [reply MobSF/rest_api.py]: Should be good!

File: MobSF/scanning.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ajinabraham) [reply MobSF/scanning.py]: Not empty , I am asking to move it inside the dict like

```
data = {
            'url': 'StaticAnalyzer/?name={}&type=apk&checksum={}'.format(self.file_name, md5),
            'status': 'success'
             'hash': ....,
             'file_name: .....,
             .............etc
        }
```

[the remark this answers is not in the document]
Comment (magaofei) [reply MobSF/scanning.py]: ok

[the remark this answers is not in the document]
Comment (magaofei) [reply MobSF/scanning.py]: @ajinabraham modified

File: MobSF/forms.py [hunk 5]
```diff
@@ -2,3 +2,17 @@
...
+
+
+class FormUtil(object):
+
+    def __init__(self, form):
+        self.form = form
+
+    @staticmethod
+    def errors_message(form):
+        return '; '.join([v for k, v in form.errors.items() for v in v])
```
[the remark this answers is not in the document]
Comment (magaofei) [reply MobSF/forms.py]: @ajinabraham I modified the new one

**Comment by ajinabraham** [comment]
@magaofei Good to merge!

**Pull request merged**